ui/src/ModalFormComponents/Select.tsx

```diff
@@ -15,59 +15,143 @@
  *   limitations under the License.
  */
 
-import React, {ReactNode, useEffect, useState} from 'react';
+import React, {ReactNode, useEffect, useRef, useState} from 'react';
+import {debounce} from '../Utils';
 
 import './Select.scss';
 
+const DEFAULT_CURRENT_INDEX = 0;
+
 export interface OptionType {
     value: unknown;
+    ariaLabel: string;
     displayValue: ReactNode | string;
 }
 
 interface Props {
+    ariaLabel: string;
     selectedOption: OptionType;
     options: Array<OptionType>;
     onChange: (selectedOption: OptionType) => void;
 }
 
-const Select = ({ options, selectedOption, onChange }: Props): JSX.Element => {
+const Select = ({ ariaLabel, options, selectedOption, onChange }: Props): JSX.Element => {
     const [dropdownToggle, setDropdownToggle] = useState<boolean>(false);
     const [currentOption, setCurrentOption] = useState<OptionType>(selectedOption);
+    const dropdownToggleElement = useRef<HTMLButtonElement>(null);
+    const dropdownElement = useRef<HTMLUListElement>(null);
+    const [currentIndex, setCurrentIndex] = useState<number>(
+        DEFAULT_CURRENT_INDEX
+    );
+    const [upKey, downKey, enterKey] = [38, 40, 13];
+
+    useEffect(() => {
+        const currentOptionIndex = options.map(option => JSON.stringify(option.value))
+            .indexOf(JSON.stringify(currentOption.value));
```

Comment:
Is there a cleaner way to do this?

Reply:
Not really that I know of. The value property we wanted to keep flexible so it could be a number, object, etc.